packages/app/src/app/utils/teams.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  SubscriptionType,
+  TeamFragmentDashboardFragment,
+  TeamMemberAuthorization,
+} from 'app/graphql/types';
+
+export const getUpgradeableTeams = ({
+  teams,
+  personalWorkspaceId,
+  userId,
+}: {
+  teams: TeamFragmentDashboardFragment[];
+  personalWorkspaceId: string | undefined;
+  userId: string | undefined;
+}) =>
+  teams.filter(team => {
+    if (
+      team.id === personalWorkspaceId ||
+      team.subscription?.type === SubscriptionType.TeamPro
+    ) {
+      return false;
+    }
+
+    const billingManagers = team.userAuthorizations
+      .filter(
```

Comment:
I think this is the same as my comment above, we can find the user first and check if it has the right authorization rather than finding members with the right authorization and then checking if the current user is part of that list.